Improve soroban setup instructions. (#410)
stellar/stellar-docs: docs/smart-contracts/getting-started/setup.mdx

## Install Rust

### Linux, macOS, or other Unix-like OS

If you use macOS, Linux, or another Unix-like OS, the simplest method to install a Rust toolchain is to install `rustup`. Install `rustup` with the following command.

```bash
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```

### Windows

On Windows, download and run [rustup-init.exe](https://static.rust-lang.org/rustup/dist/i686-pc-windows-gnu/rustup-init.exe). You can continue with the default settings by pressing Enter.

:::tip

The soroban CLI uses emojis. To properly render them on Windows, we recommend using [Windows Terminal](https://learn.microsoft.com/en-us/windows/terminal/). See [how to install Windows Terminal](https://learn.microsoft.com/en-us/windows/terminal/install) on Microsoft Learn.

:::

If you're already using [WSL](https://learn.microsoft.com/en-us/windows/wsl/install), you can also follow the instructions for Linux.

### Other

> For other methods of installing [Rust], check out: https://www.rust-lang.org/tools/install

Reviewer: Small tweak to remove language we'll use elsewhere when referring to git operations.
```suggestion
For other methods of installing [Rust], see: https://www.rust-lang.org/tools/install
```

To configure your CLI to interact with Testnet, run the following command:

<Tabs>
  <TabItem value="nix" label="macOS/Linux">

```bash
soroban network add \
  --global testnet \
  --rpc-url https://soroban-testnet.stellar.org:443 \
  --network-passphrase "Test SDF Network ; September 2015"
```

  </TabItem>
  <TabItem value="windows" label="Windows">

```bash
soroban network add --global testnet --rpc-url https://soroban-testnet.stellar.org:443 --network-passphrase "Test SDF Network ; September 2015"

> ```

Reviewer: Actually, quick thought: We should be able to use the `^` on windows for line continuation. I don't have windows setup to test it, but I know you do. Worth checking if we can preserve the multiline view for easy reading. But we'd need to check it on cmd.exe, powershell, and windows terminal I guess. If they don't all work the same way, then yeah lets keep it as is.

Like the Network configs, the `--global` means that the identity gets stored in `~/.config/soroban/identity/alice.toml`. You can omit the `--global` flag to store the identity in your project's `.soroban/identity` folder instead.

> By default, `soroban keys generate` will fund your account using [Friendbot](https://developers.stellar.org/docs/fundamentals-and-concepts/testnet-and-pubnet#friendbot). If you don't want this behavior, run this command with `--no-fund`.

Reviewer: Small tweak to reduce the chance this is misinterpreted to mean you need to run an additional command with the option:
```suggestion
By default, `soroban keys generate` will fund your account using [Friendbot](https://developers.stellar.org/docs/fundamentals-and-concepts/testnet-and-pubnet#friendbot). If you don't want this behavior, append `--no-fund` to the command when running it.
```
